controller/userController.js

```diff
@@ -0,0 +1,272 @@
+    if (foundData) {
+      res.send(foundData);
+    } else {
+      res.send(404);
+    }
+  } catch (error) {
+    console.log(error);
+    res.json({
+      status: 500,
+      message: error,
+    });
+  }
+};
+const deluserData = async (req, res) => {
+  try {
+    let decoded = req.user;
+    let founduser = await data.findOne({ _id: decoded.registration });
+    if (founduser) {
+      await data.deleteOne({ _id: decoded.registration });
+      res.send("data deleted");
+    } else {
+      console.log("data does not exist");
+    }
+  } catch (error) {
+    console.log(error);
+    res.json({
+      status: 500,
+      message: error,
+    });
+  }
+};
+const paginateddataReturn = async (req, res) => {
+  try {
+    const { page = req.query.id, limit = 10 } = req.query;
+    const posts = await data
+      .find()
+      .limit(limit * 1)
+      .skip((page - 1) * limit);
+
+    const count = await data.countDocuments();
+    res.send(count);
+    res.json({
+      posts,
+      totalPages: Math.ceil(count / limit),
+      currentPage: page,
+    });
+  } catch (err) {
+    console.error(err.message);
+  }
+};
+const addressCreate = async (req, res) => {
+  try {
+    let uid = req.user;
+
+    let createdadta = await userAddress.create({
+      user_id: uid.user_id,
+      address: req.body.address,
+      city: req.body.city,
+      state: req.body.state,
+      phonenumber: req.body.phonenumber,
+      pincode: req.body.pincode,
+    });
+
+    res.send(createdadta);
+  } catch (error) {
+    console.log(error);
+    res.json({
+      status: 0,
+      message: error,
+    });
+  }
+};
+const deleteAddress = async (req, res) => {
+  try {
+    let uid = req.user;
+    let findaddressdata = await userAddress.findOne({
+      user_id: uid.registration,
+    });
+    if (!findaddressdata) {
+      res.send("invalid address");
+    } else {
+      await userAddress.deleteOne({ address: findaddressdata.address });
+      res.send("address deleted");
+    }
+  } catch (error) {
+    console.log(error);
+    res.json({
+      status: 0,
+      message: error,
+    });
+  }
+};
+const passwordResetgenrator = async (req, res) => {
+  try {
+    let userdata = await data.findOne({ email: req.body.email });
+    if (!userdata) {
+      res.send("please enter valid email");
+    } else {
+      let token = jwt.sign({ registration: userdata._id }, process.env.key, {
+        expiresIn: "1800s",
+      });const msg = {
+        to: 'abaloni02@gmail.com', 
+        from: 'anmol@excellencetechnologies.in', 
+        subject: 'password reset',
+        text: 'you requested to reset the password',
+        html: `<link>http://localhost:3000/user/verify_reset_password/${token}</link>`,
+      }
+     await sgMail
+    .send(msg)
+    .then(() => {
+      console.log('Email sent')
+    })
+      await userTokengenrator.create({
+        userid : userdata._id,
+        access_token :token
+      })
+
+
+      res.send(token);
```

Comment:
do not return token here , tell the user that password reset link has been shared